src/main/java/com/ontotext/trree/geosparql/util/GeoSparqlUtils.java

```diff
@@ -20,6 +21,29 @@
  * @since 01 Oct 2015.
  */
 public class GeoSparqlUtils {
+    public static void migrateConfig(Path pluginDataDir, Logger logger) {
+        GeoSparqlConfig config = new GeoSparqlConfig();
+
+        Path configPath = GeoSparqlConfig.resolveConfigPath(pluginDataDir);
+        if (!Files.exists(configPath)) {
+            Path legacyConfigPath = GeoSparqlConfig.resolveLegacyConfigPath(pluginDataDir);
+            if (Files.isReadable(legacyConfigPath)) {
+                try (FileInputStream in = new FileInputStream(legacyConfigPath.toFile())) {
+                    Properties properties = new Properties();
+                    properties.load(in);
+                    config.setFromProperties(properties);
+                    if (config.isEnabled()) {
+                        logger.info("Upgrade detected. Index is created with previous GraphDB version. Recreate index");
```

Comment:
Change this to "Detected incompatible index from a previous version. Please rebuild the index manually."